clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -128,15 +138,32 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 				return c.CoreV1().Pods(p.selector.Namespace).Watch(context.TODO(), options)
 			},
 		},
-		p.checkPod,
+		p.addEvent,
 	)
+	 go p.processEvents()
 	return informer.StartAndSync(i, p.stopCh, informerSyncTimeout)
 }
 
+func (p *podStartupLatencyMeasurement) addEvent(_, obj interface{}) {
+	p.eventCh <- &eventData{obj, time.Now()}
```

Comment:
Good catch.

I would go one step further here and in case the channel is already full log some error or something. So that we know we won't hit this issue.